code/game/turfs/simulated/floor.dm

```diff
@@ -120,30 +120,31 @@ var/list/icons_to_ignore_at_floor_init = list("damaged1","damaged2","damaged3","
 /turf/simulated/floor/proc/make_plating()
 	return ChangeTurf(/turf/simulated/floor/plating)
 
-/turf/simulated/floor/ChangeTurf(turf/simulated/floor/T, defer_change = FALSE, keep_icon = TRUE)
-	if(!istype(src,/turf/simulated/floor)) return ..() //fucking turfs switch the fucking src of the fucking running procs
-	if(!ispath(T,/turf/simulated/floor)) return ..()
-	var/old_icon = icon_regular_floor
-	var/old_plating = icon_plating
-	var/old_dir = dir
+/turf/simulated/floor/ChangeTurf(turf/simulated/floor/T, defer_change = FALSE, ignore_air = FALSE)

```

Comment:
You don't seem to change the way this function is called in the map loader,
`modules/awaymissions/maploader/reader.dm`.
That's the one piece of code that uses all 3 of these arguments. - `T.ChangeTurf(path, 1, 0)`
Are you sure the effects of this change will not adversely affect the maploader?

Reply:
I just ran another test: both ruins and away missions load totally fine (although I did forget to remove the second false parameter). /tg/ got rid of the keep_icons a while back, and I have no idea what we even used it for. My guess is something like icon smoothing, which is handled properly now anyway.